src/github_api.py

```diff
@@ -56,3 +58,235 @@ def get_issues() -> dict:
     else:
         logger.info(f"Github Response Code: {response.status_code}")
         return response.json()
+
+def get_fields_info() -> dict:
+  """
+  GraphQLを使ってGitHubProjectからIssue情報を取得する
+  """
+  headers = {"Authorization": f"bearer {GH_TOKEN}"}
+  url = "https://api.github.com/graphql"
+  query_template = """
+    {
+      node(id: "{{PROJECT_ID}}") {
+        ... on ProjectV2 {
+          fields(first: 100) { # 10は取得するフィールドの数、必要に応じて変更してください
+            nodes{
+                ... on ProjectV2Field{
+                    id
+                    name
+                }
+            }
+            edges {
+              node {
+                ... on ProjectV2SingleSelectField {
+                  id
+                  name
+                  options {
+                    id
+                    name
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  """
+  query = Template(query_template).render(PROJECT_ID=GH_PROJECT_ID)
+  response = requests.post(url, json={"query": query}, headers=headers)
+  if response.status_code != 200 or "errors" in response.json().keys():
```

Comment:
```suggestion
  if response.status_code != 200 or "errors" in response.json():
```
で良いかなと思いますが、
確かGithub APIって200なのにerrorsを投げるとかいう、テメェ！！ってなる仕様だった気がしますが、
確かそうでしたよね？（それならもうしょうがないんですけどね）

あと、200以外の条件で落ちることが、割とたくさんある気がします。
それを全部死にましたで投げると、死んだ時、なんだこれってなるんで

- 400 : デグレった？
- 403 : tokenが(expiredとかで)死んだ？
- 404 : APIのendpoint変わった？
- 5xx : 放置してりゃ取りあえずなんとかなるやろ

くらいの感じで振り分けたいです。

Reply:
まぁでもresponse.jsonで全部わかるからええやろ。みたいなのでも全然良きです。

Reply:
ですね(projectIDを適当にセットした際のもんです)
<img width="1211" alt="スクリーンショット 2024-03-09 13 41 48" src="https://github.com/TechTeam-org/public-tasks/assets/159105331/161f19e3-e45d-489a-ac55-4e9ed4dd389e">

Reply:
内部品質の向上でやるのは全然良きかなと思います。
ただこれ別タスクで切り出しても良いですか？
github_api.pyに書かれている全てに言えることなので、横断的に治したいです。